includes/wizards/class-setup-wizard.php

```diff
@@ -504,6 +504,35 @@ public function api_update_theme_with_mods( $request ) {
 				continue;
 			}
 
+			if ( substr_compare( $key, '_all_posts', -strlen( '_all_posts' ) ) === 0 ) {
+				$post_ids = get_posts(
+					[
+						'posts_per_page' => -1,
```

Comment:
This `-1` query could be showstopping for sites with a lot of posts. And even though the new settings are unlikely to be changed much after a site launches, plenty of sites with lots of migrated content might want to change them pre-launch. 

A two-fold suggestion:

1. Batch the query and execution with a `do...while` loop so we're only processing only ~100 posts at a time, and if there are more than 10,000 posts or so on the site, warn that this could take a long time and give the option to bail.
2. Optimize the query so we fetch posts with a `_wp_page_template` meta value that doesn't match the given value.

Reply:
Good ideas! I've added batching (25473e75c5069981416af0cbe7fb281baee446a1), meta value comparison (8456328ae111b02b1cfbfe656baddb60561aa8f1), and a warning notice if the site has more than a thousand posts (16e043b727d00edbe1f9e82444bc0a72dde71e07):

<img width="703" alt="image" src="https://github.com/Automattic/newspack-plugin/assets/7383192/5dd2f906-59e1-4a3e-8740-2e2ff7f04420">